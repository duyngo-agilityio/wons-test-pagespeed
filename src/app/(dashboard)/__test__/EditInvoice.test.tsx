import { render, waitFor } from '@testing-library/react';

import EditInvoicePage from '@/app/(dashboard)/edit-invoice/page';

jest.mock('@/ui', () => ({
  EditInvoice: () => <div data-testid="edit-invoice">Edit Invoice</div>,
}));

jest.mock('next/navigation', () => ({
  ...jest.requireActual('next/navigation'),
  notFound: jest.fn(),
}));

describe('Edit Invoice Page render', () => {
  it('should render and match with snapshot', async () => {
    const { container } = render(
      await EditInvoicePage({ searchParams: { id: 1 } }),
    );

    await waitFor(() => {
      expect(container).toBeInTheDocument();
    });

    expect(container).toMatchSnapshot();
  });
});
