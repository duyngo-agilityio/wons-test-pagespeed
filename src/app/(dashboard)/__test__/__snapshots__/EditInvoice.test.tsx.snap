// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit Invoice Page render should render and match with snapshot 1`] = `
<div>
  <div
    data-testid="edit-invoice"
  >
    Edit Invoice
  </div>
</div>
`;
